src/code42cli/cmds/watchlists.py

```diff
@@ -0,0 +1,250 @@
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def watchlists(state):
+    """Manage watchlist user memberships."""
+    pass
+
+
+@watchlists.command("list")
+@format_option
+@sdk_options()
+def _list(state, format):
+    """List all watchlists."""
+    pages = state.sdk.watchlists.get_all()
+    dfs = (DataFrame(page["watchlists"]) for page in pages)
+    formatter = DataFrameOutputFormatter(format)
+    formatter.echo_formatted_dataframes(dfs)
+
+
+@watchlists.command()
+@click.option(
+    "--watchlist-id",
+    help="ID of the watchlist.",
+)
+@click.option(
+    "--watchlist-type",
+    type=click.Choice(WatchlistType.choices()),
+    help="Type of watchlist to remove user from. For CUSTOM watchlists, --watchlist-id is required.",
+    cls=incompatible_with("watchlist_id"),
+)
+@format_option
+@sdk_options()
+def list_users(state, watchlist_type, watchlist_id, format):
+    """List all users on a given watchlist."""
+    if not watchlist_id and not watchlist_type:
+        raise click.ClickException("--watchlist-id OR --watchlist-type is required.")
+    if watchlist_type:
+        watchlist_id = state.sdk.watchlists._watchlists_service.watchlist_type_id_map[
+            watchlist_type
+        ]
+    pages = state.sdk.watchlists.get_all_included_users(watchlist_id)
+    dfs = (DataFrame(page["includedUsers"]) for page in pages)
+    formatter = DataFrameOutputFormatter(format)
+    formatter.echo_formatted_dataframes(dfs)
+
+
+@watchlists.command()
+@click.option(
+    "--watchlist-id",
+    help="ID of the watchlist.",
+)
+@click.option(
+    "--watchlist-type",
+    type=click.Choice(WatchlistType.choices()),
+    help="Type of watchlist to remove user from. For CUSTOM watchlists, --watchlist-id is required.",
+    cls=incompatible_with("watchlist_id"),
+)
+@click.argument("user", metavar="USER_ID|USERNAME")
+@sdk_options()
+def add(state, watchlist_id, watchlist_type, user):
+    if not watchlist_id and not watchlist_type:
+        raise click.ClickException("--watchlist-id OR --watchlist-type is required.")
+    try:
+        user = int(user)
+    except ValueError:
+        # assume username if `user` is not an int
+        user = state.sdk.userriskprofile.get_by_username(user)["userId"]
+    try:
+        if watchlist_id:
+            state.sdk.watchlists.add_included_users_by_watchlist_id(user, watchlist_id)
+        elif watchlist_type:
+            state.sdk.watchlists.add_included_users_by_watchlist_type(
+                user, watchlist_type
+            )
+    except Py42WatchlistNotFound:
+        raise
+    except Py42NotFoundError:
+        raise Code42CLIError(f"User ID {user} not found.")
+
+
+@watchlists.command()
+@click.option("--watchlist-id", help="ID of the watchlist.")
+@click.option(
+    "--watchlist-type",
+    type=click.Choice(WatchlistType.choices()),
+    help="Type of watchlist to remove user from. For CUSTOM watchlists, --watchlist-id is required.",
```

Comment:
Do we want to talk about custom watchlists before they're released? Or does that not really matter

Reply:
I suppose I can leave this off for now.